base_layer/core/src/proof_of_work/blake_pow.rs

```diff
@@ -43,6 +43,8 @@ pub struct BlakePow;
 
 impl BlakePow {
     /// A simple miner. It starts at nonce = 0 and iterates until it finds a header hash that meets the desired target
+    // ToDo convert to future, with ability to break function. We need to be able to stop this if we receive a mined
+    // block
```

Comment:
For the record, This was really just a function to help for tests and wasn't ever intended to be used in production.

Reply:
The only way I can think of is too move all of out test code into a "test" crate
We can then import this as a dev dependancy